types.ts

```diff
@@ -546,6 +560,12 @@ export interface CustomAuthentication extends BaseAuthentication {
    * replacement inside the constructed network request.
    */
   params: CustomAuthParameter[];
+  /**
+   * An optional header hex digest transform that will be performed once template replacement is complete. 
+   * The header {@link CustomAuthHeaderDigestTransform.header} will be transformed using the specified hex
+   * digest {@link CustomAuthHeaderDigestTransform.algorithm}.
+   */
+  digestTransform?: CustomAuthHeaderDigestTransform
```

Comment:
I wonder if we should follow a pattern similar to `postSetup`, where we have an array of transform steps, with the only implemented step type at the moment being this one. That would allow devs to use multiple digest transforms on a single request and give us a place to hang future oddball transforms that are needed.